packages/core/workers/src/WorkerFarm.js

```diff
@@ -502,15 +502,21 @@ export default class WorkerFarm extends EventEmitter {
     let promises = [];
     for (let worker of this.workers.values()) {
       promises.push(
-        new Promise((resolve, reject) => {
-          worker.call({
-            method,
-            args,
-            resolve,
-            reject,
-            retries: 0,
-          });
-        }),
+        (worker.ready
+          ? Promise.resolve()
+          : new Promise(res => worker.once('ready', res))
+        ).then(
+          () =>
+            new Promise((resolve, reject) =>
+              worker.call({
```

Comment:
could this waiting logic be moved into `worker.call()` itself?

Reply:
We'd have to make sure that this call here
https://github.com/parcel-bundler/parcel/blob/7fb0d22dee6ddc8554c1849d4e78ade23e09ba47/packages/core/workers/src/Worker.js#L90-L101

doesn't get this treatment

Reply:
Done with `skipReadyCheck` in the `.call` options.